File: 프로그래머스/lv0/120880. 특이한 정렬/특이한 정렬.java

import java.util.*;

class Solution {
    public ArrayList<Integer> solution(int[] numlist, int n) {

        ArrayList<Integer> intList = new ArrayList<>();

        for (int i = 0; i < numlist.length; i++) {
            intList.add(Integer.valueOf(numlist[i]));
        }
        intList.sort(new Comparator<Integer>() {
            @Override
            public int compare(Integer o1, Integer o2) {
                int o1Int = o1.intValue();
                int o2Int = o2.intValue();

                if (Math.abs(o1Int-n) < Math.abs(o2Int-n)) return -1;
                else if (Math.abs(o1Int-n) > Math.abs(o2Int-n)) return 1;
                else return o2-o1;
            }
        });

        System.out.println(intList);

        return intList;
    }
}

File: 프로그래머스/lv0/120880. 특이한 정렬/특이한 정렬.cs
using System;
using System.Linq;
public class Solution {
    public int[] solution(int[] numlist, int n) {
        int[] answer = new int[] {};
        return numlist.OrderBy(a => Math.Abs(a - n)).ThenByDescending(a => a).ToArray();
    }
}